packages/libllvm/tools-clang-lib-Driver-ToolChains-Linux.cpp.patch

```diff
@@ -1,18 +1,40 @@
---- ./tools/clang/lib/Driver/ToolChains/Linux.cpp.orig	2018-07-30 19:44:13.000000000 +0000
-+++ ./tools/clang/lib/Driver/ToolChains/Linux.cpp	2019-03-14 03:42:40.783434246 +0000
-@@ -291,7 +291,10 @@
-   // possible permutations of these directories, and seeing which ones it added
-   // to the link paths.
-   path_list &Paths = getFilePaths();
--
-+  if(IsAndroid) {
-+    addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/lib", Paths);
-+	ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/lib");
-+  }
+--- tools/clang/lib/Driver/ToolChains/Linux.cpp.orig	2019-03-28 00:40:14.508554643 +0000
++++ ./tools/clang/lib/Driver/ToolChains/Linux.cpp	2019-03-28 00:46:12.983564436 +0000
+@@ -314,6 +314,33 @@
+ 
    const std::string OSLibDir = getOSLibDir(Triple, Args);
    const std::string MultiarchTriple = getMultiarchTriple(D, Triple, SysRoot);
++  
++  if(IsAndroid) {
++    if (MultiarchTriple == llvm::sys::getDefaultTargetTriple()) {
++      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/lib", Paths);
++    }
++    if (Arch == llvm::Triple::arm || Arch == llvm::Triple::thumb) {
++      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/arm-linux-androideabi/lib", Paths);
++      addPathIfExists(D, SysRoot + "/system/lib", Paths);
++      ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/arm-linux-androideabi/lib");
```

Comment:
Since this has been merged, I dug a bit deeper: is there any point to adding this rpath for the arch-dependent lib? All I see is a bunch of object files and a static lib in these arch-dependent directories installed from the ndk-multilib package, so this rpath currently won't find any shared libraries.

Is there some plan to start putting shared libraries in there eventually or what?

Reply:
There no plan with putting there any shared libraries or introducing multilib prefix.
Just useful thing for those who cross-compile and run stuff on device as it may require not just ndk-sysroot but additional deps which may be install by user manually to e.g. `$PREFIX/arm-linux-androideabi`.

Reply:
OK, then I will submit a pull to remove that additional rpath if not cross-compiling, as I notice it adds that aarch64 directory even if natively building on aarch64.